satpy/readers/netcdf_utils.py

```diff
@@ -187,27 +191,44 @@ def collect_cache_vars(self, cache_vars, obj):
 
     def __getitem__(self, key):
         """Get item for given key."""
+        if key in ('/attr', '/attrs'):
+            return self._get_global_attrs()
         val = self.file_content[key]
         if isinstance(val, netCDF4.Variable):
-            if key in self.cached_file_content:
-                return self.cached_file_content[key]
-            # these datasets are closed and inaccessible when the file is
-            # closed, need to reopen
-            # TODO: Handle HDF4 versus NetCDF3 versus NetCDF4
-            parts = key.rsplit('/', 1)
-            if len(parts) == 2:
-                group, key = parts
-            else:
-                group = None
-            if self.file_handle is not None:
-                val = self._get_var_from_filehandle(group, key)
-            else:
-                val = self._get_var_from_xr(group, key)
+            return self._get_variable(key, val)
         elif isinstance(val, netCDF4.Group):
```

Comment:
Just to make codefactor.io happy

```suggestion
        if isinstance(val, netCDF4.Group):
```